src/Shared/FileMatcher.cs

```diff
@@ -836,6 +837,20 @@ struct RecursionState
             Dictionary<string, List<RecursionState>> searchesToExcludeInSubdirs,
             TaskOptions taskOptions)
         {
+#if NET6_0
```

Comment:
As a rule, never use exact-match checks in ifdefs. If you mean "if this API is available" you could say

```suggestion
#if NET6_0_OR_GREATER
```

but the preferred approach is to create a `FEATURE_*` flag and set it when relevant, checking it here. That's nice because it allows future maintainers to more easily understand the nature of the conditional inclusion: it was like this because `LinkTarget` was not available on .NET 4.7.2.

Reply:
I can certainly do NET6_0_OR_GREATER, but I'm a bit hesitant about a new FEATURE_* for a few reasons. First, making MSBuild logic for "or greater" isn't easy for this sort of thing. I'd have to parse out the version from the target framework then do a version comparison with 6. That messiness is presumably why our other flags that would use NET6_0_OR_GREATER-dependent feature flags use NET6_0_OR_GREATER rather than making new flags, and I do like consistency.

Reply:
You can put it in this section:

https://github.com/dotnet/msbuild/blob/bfd80ab1692721767c3fa8c0d4bcd35254489086/src/Directory.BeforeCommon.targets#L87-L93

Reply:
This is the standard approach in our codebase. It does not seem important to me to support a theoretical future downgrade to an end-of-life target, and I don't think it's "lying" to say "when building for the current .NET Core TFM, include this code that is present in today's version of that TFM".